torchvision/datasets/mnist.py

```diff
@@ -286,7 +286,7 @@ class EMNIST(MNIST):
         'mnist': list(string.digits),
     }
 
-    def __init__(self, root: str, split: str, **kwargs: Any) -> None:
+    def __init__(self, root: str, split: str = "mnist", **kwargs: Any) -> None:
```

Comment:
Similar to other `MNIST` variants, we add a default value for the split. This makes testing a lot easier, since we otherwise need some special handling for iterating over a required parameter. Still, this is fully BC. I've chosen `split="mnist"` as default, because, as the name implies, has the size as the default `MNIST` dataset.

Reply:
Could we instead just specify the `split` in the `CONFIGS` class attribute of the test class?

Reply:
I'm already doing that. The problem is for tests that don't run against all configs, the infrastructure detects the default config from the default values in the dataset constructor. Since `split` has none (without this patch) it won't get picked up. Instead of this patch, we could offer a mechanism to manually overwrite the default config. 

Although this is not hard to implement, IMO having a default value is actually a good thing. The other `MNIST` datasets are also opinionated about which split is loaded by default.

Reply:
> the infrastructure detects the default config from the default values in the dataset constructor

The docs about `CONFIG` say:

> The first element will be used as default configuration

Is this still up to date?

Reply:
Good catch! No, this is no longer true. The behavior changed in #3450. The rationale was that I wanted to make sure to always pass a "complete" config to `inject_fake_data`. I had a case (but can't find it now) where the `inject_fake_data` depended on `"foo"` while part of the configs only iterated over `"bar"`. Thus we now get the default config from the default arguments of the dataset constructor

https://github.com/pytorch/vision/blob/bb2805a669b197582b796f0d86ef8932e1a4396a/test/datasets_utils.py#L360-L366

and use this as base for the user specified config 

https://github.com/pytorch/vision/blob/bb2805a669b197582b796f0d86ef8932e1a4396a/test/datasets_utils.py#L311-L314

Reply:
I've changed my mind a little about this: Although I'm still convinced that a default value for `split` makes sense, I think the test architecture should be flexible enough to handle this case. I'll send a PR.